src/main/java/com/programmers/voucher/domain/Voucher.java

```diff
@@ -1,16 +1,64 @@
 package com.programmers.voucher.domain;
 
-import java.time.LocalDate;
+import java.time.LocalDateTime;
 import java.util.UUID;
 
-public interface Voucher {
-    UUID getVoucherId();
+public class Voucher {
 
-    Discount getDiscount();
+    private static final String VOUCHER_NULL_MESSAGE = "[ERROR] 유효하지 않은 Voucher Id 입니다. (null)";
+    private static final String EXPIRED_VOUCHER_MESSAGE = "[ERROR] 기간이 만료된 Voucher 입니다.";
+    private static int VOUCHER_EXPIRATION_POLICY = 7;
 
-    LocalDate getCreatedDate();
+    private final UUID voucherId;
+    private final Discount discount;
+    private final LocalDateTime createdAt;
+    private final LocalDateTime expiration;
 
-    LocalDate getExpirationDate();
+    public Voucher(UUID voucherId, Discount discount, LocalDateTime createdAt) {
+        validateVoucherId(voucherId);
+        this.voucherId = voucherId;
+        this.discount = discount;
+        this.createdAt = createdAt;
+        this.expiration = applyExpiration();
+    }
 
-    long discount(long beforeDiscount);
+    public Voucher(UUID voucherId, Discount discount, LocalDateTime createdAt, LocalDateTime expiration) {
+        this.voucherId = voucherId;
+        this.discount = discount;
+        this.createdAt = createdAt;
+        this.expiration = expiration;
+    }
+
+    public long discount(long itemPrice) {
+        checkVoucherExpiration();
+        return discount.applyDiscount(itemPrice);
+    }
```

Comment:
```suggestion
    public long discountWith(long itemPrice, LocatDateTime usedAt) {
        checkVoucherExpiration(usedAt);
        return discount.applyDiscount(itemPrice);
    }
```

바우처의 유효기간을 검증한다고 생각해볼게요. 내부에 `now()`를 호출하는 것 처럼 제어할 수 없는 값이 존재하면 검증하기 어려울 것 같아요. 예전에는 만료되지 않았던 테스트 코드가 시간이 지나서 만료될 수도 있을 것 같아요.